examples/connect-app-teleop-react/server.js

```diff
@@ -0,0 +1,393 @@
+import express from 'express';
+import session from 'express-session';
+import nocache from 'nocache';
+
+const serverPort = 9000;
+
+// configure Express app and install the JSON middleware for parsing JSON bodies
+const app = express();
+
+app.use(bodyParser.json());
+
+app.use(cookieParser());
+
+// configure CORS
+app.use(
+  cors({
+    origin: true,
+    credentials: true,
+  })
+);
+
+app.use(nocache());
+
+// configure sessions
+app.use(
+  session({
+    secret: '1234567890',
+    resave: false,
+    saveUninitialized: false,
+    cookie: {
+      secure: true,
+      httpOnly: true,
+      maxAge: 3600000,
+      sameSite: 'lax',
+    },
+  })
+);
+
+// use routesr
+app.use(express.static('./dist'));
+
+if (process.env.VITE_AUTH_CLIENT_ID && process.env.VITE_AUTH_CLIENT_SECRET) {
+  console.log(
+    'running with FusionAuth for client',
+    process.env.VITE_AUTH_CLIENT_ID
+  );
+
+  const config = {
+    clientId: process.env.VITE_AUTH_CLIENT_ID,
+    clientSecret: process.env.VITE_AUTH_CLIENT_SECRET,
+    fusionAuthBaseUrl: 'https://viam-prod.fusionauth.io',
+  };
+
+  app.get('/app/callback', async (req, res) => {
+    console.log('accepting request for token exchange');
+    const code = req.query.code;
+    const codeVerifier = req.cookies.codeVerifier;
+    const redirect_uri = `${req.protocol}://${req.get('host')}/app/callback`;
+
+    try {
+      // POST request to /oauth2/token endpoint
+      const fusionAuthResponse = await fusionAuthClient('/oauth2/token', {
+        method: 'POST',
+        headers: {
+          'Content-Type': 'application/x-www-form-urlencoded',
+        },
+        body: getFormURLEncodedPayload({
+          client_id: config.clientId,
+          client_secret: config.clientSecret,
+          code: code,
+          code_verifier: codeVerifier,
+          grant_type: 'authorization_code',
+          redirect_uri: redirect_uri,
+        }),
+      });
+
+      const { access_token, id_token, refresh_token, expires_in } =
+        fusionAuthResponse;
+      if (!(access_token && refresh_token)) {
+        console.log('Either refresh token or access token is missing.');
+        res.sendStatus(503);
+        return;
+      }
+
+      console.log('saving tokens as cookies');
+      // save tokens as cookies
+      // the access token will be readable by the web app and susceptible
+      // to XSS attacks. It may be better to put this in localstorage or
+      // to make the web server a proxy to viam.
+      setReadableCookie(res, 'app.at', access_token);
+      setSecureCookie(res, 'app.rt', refresh_token);
+
+      const expires_in_ms = expires_in * 1000;
+      setReadableCookie(res, 'app.at_exp', (Date.now() + expires_in_ms) / 1000);
+      setReadableCookie(res, 'codeVerifier', '', 0);
+      setReadableCookie(res, 'app.idt', id_token);
+
+      const redirectUrl = generateRedirectUrlFromState(req);
+
+      res.redirect(redirectUrl);
+    } catch (error) {
+      res.status(500).send(error);
+    }
```

Comment:
IIRC returning a 500 if a handler throws is default express behavior, is the wrapping `try` necessary? Or is this something silly like express doesn't actually check the return of the async handler?

If it's necessary, could we scope down the `try/catch` to the call(s) we're actually trying to guard? Big `try` blocks make me nervous and I find it harder to discern the original developer's intent when reading them

Reply:
this is all copy and pasted from https://github.com/FusionAuth/fusionauth-javascript-sdk-express/ so I'd rather not change this unless you feel strongly

Reply:
Got it, didn't know it was copy pasted. Mind leaving a comment to that effect with the source? No need to mess with it if it's not our code

Reply:
yep, it's already at the top with license attribution as well

Reply:
the change i made was concatenated everything to one file for ease of use